src/proxy_app/router_core.py

```
@@ -637,91 +649,141 @@ def __init__(self, config_path: str = "config/router_config.yaml"):
+                success=success,
+                response_time_ms=response_time_ms,
+                error_reason=error_category.value if error_category else None,
+                tokens_per_second=tps,
+                input_tokens=input_tokens,
+                output_tokens=output_tokens,
+            )
+
+            if tps is not None:
+                telemetry.record_tps(provider, model, tps)
+
+        except Exception as telemetry_err:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)